feat(event-gateway): Clarifications and more info on authentication for KEG (#6491)
Kong/developer.konghq.com: app/event-gateway/entities/virtual-cluster.md

      Requires clients to present a trusted TLS certificate during the TLS handshake.
      The client certificate is verified against a [TLS trust bundle](/event-gateway/entities/tls-trust-bundle/).
      The certificate's principal (for example, the Common Name) can be used to enforce access control with [ACL policies](/event-gateway/policies/acl/).
      <br><br>
      See [Configure mTLS client authentication](/event-gateway/configure-mtls-client-authentication/) for a full walkthrough, including how to configure principal extraction.
      <br><br>

>       mTLS is also supported for the gateway-to-backend connection; configure it in the [backend cluster's TLS settings](/event-gateway/entities/backend-cluster/).

Reviewer: ```suggestion
      mTLS is also supported for the gateway-to-backend connection; configure it in the [backend cluster's TLS settings](/event-gateway/entities/backend-cluster/#schema-backend-cluster-tls).
```
I think this link might be more helpful?

Behind the scenes, this maps to `team-a-orders` on the actual Kafka cluster. The client doesn't need to know about or manage the `team-a-` prefix.
This enables transparent multitenancy, where multiple teams can share the same Kafka cluster without needing to manually prefix every topic and consumer group name in their applications.

> When clients send fetch requests, the gateway passes the consumer group name to the backend unchanged, unless a namespace prefix is configured.

Reviewer 2: That's not quite technically correct.

When client uses consumer group they use [OffsetFetch](https://kafka.dyszkiewicz.me/?tab=messages&message=OffsetFetch) to fetch current offsets for topic partition, then it sends [Fetch](https://kafka.dyszkiewicz.me/?tab=messages&message=Fetch) to get actuall data and then [OffsetCommit](https://kafka.dyszkiewicz.me/?tab=messages&message=OffsetCommit) to let backend know they received the data.

Maybe we should state

> When clients send messages that contain consumer group names, the gateway passes the names to the backend unchanged, unless a namespace prefix is configured

> When clients send fetch requests, the gateway passes the consumer group name to the backend unchanged, unless a namespace prefix is configured.

Author: ```suggestion
When clients send messages that contain consumer group names, the gateway passes the names to the backend unchanged, unless a namespace prefix is configured.
```

> When clients send messages that contain consumer group names, the gateway passes the names to the backend unchanged, unless a namespace prefix is configured.

Reviewer: ```suggestion
When clients send messages that contain Consumer Group names, the gateway passes the names to the backend unchanged, unless a namespace prefix is configured.
```

Kafka handles consumer group coordination and offset tracking on the backend as normal.

> Consumer group management commands such as create and delete are intercepted by the gateway and evaluated against ACL policies using the `group` resource type.

Reviewer: ```suggestion
Consumer Group management commands such as create and delete are intercepted by the gateway and evaluated against ACL policies using the `group` resource type.
```
